mods/tnt/init.lua

```diff
@@ -557,6 +561,9 @@ function tnt.register_tnt(def)
 					end
 				}
 			},
+			on_burn = function(pos)
+				minetest.set_node(pos, {name = name .. "_burning"})
```

Comment:
This would cause the tnt to explode instantly, without the ignition effects. It makes sense for on_blast, but when being next to fire I would guess the circumstances are not that different than when being next to burning gunpowder.
I don't mind changing it though. But I'll wait some time to see if there are more opinions.